### ISSUE TITLE
Every other request to /v1/users returns 503

### ISSUE THREAD

**Issue by kymppi** [rittaschool/ritta-server#76]
**Describe the bug**
When making GET request to /v1/users it returns every other time 503 and every other the actual data

**To Reproduce**
Make requests

**Expected behavior**
It should always return the data

**System information**
Docker for Ubuntu 20.04

**Comment by kymppi** [comment]
Fixed by #b837fd9f73a3d1163c02c2b5aef6102c5acb2334

**Comment by kymppi** [comment]
Thats the commit id

**Comment by GoodnessEzeokafor** [comment]
not following, how exactly was this fixed ?

**Comment by kymppi** [comment]
Thanks for the question, it might be unclear that how those changes fixed it but it was updating packages and making the configuration correct for rabbitmq. Before it just sent the event to the main-queue but after the changes it sends it for example to users-queue because other services were picking up the request to users, so users service would not receive it, it couldn't respond.

TLDR; the data fetching request didnt go to the right destination all the time

**Comment by raikasdev** [comment]
Simply: every microservice needs it own queue, or this will occur for some reason.

**Comment by GoodnessEzeokafor** [comment]
yeah, i did the same thing but had the same issue
i had to do more, for example
```
const app = await NestFactory.createMicroservice<MicroserviceOptions>(
      AppModule,
      {
        transport: Transport.RMQ,
        options: {
          urls: [RABBIT_MQ_URl],
          queue: RABBIT_MQ_QUEUES.USER,
          noAck: false,
          queueOptions: {
            durable: true
          },
        },
        logger: LOGS_LEVEL(),
      },
    );
```
I set `noAck` to `false` and `durable` to `true`
manually alcknoledge the message on my controllers
```
    const response: IResponse = await this.service.recoverPassword(payload)

    const channel = context.getChannelRef();
    const originalMsg = context.getMessage();
    channel.ack(originalMsg);
        return response


```
thanks alot, your commit set me on the right direction

**Comment by kymppi** [comment]
Glad you got it working, we chose to not use the `noAck` since it doesn't give us any benefit.
